docs/blog/2025-05-05-nx-21-release.md

```diff
@@ -0,0 +1,184 @@
+youtubeUrl: https://youtu.be/9zsncTxfcl4
+---
+
+{% callout type="deepdive" title="Nx 21 Launch Week" expanded=true %}
+
+This article is part of the Nx 21 Launch Week series:
+
+- **Nx 21 Release: Continuous tasks and Terminal UI lead the way**
+- Continuous tasks
+- Terminal UI
+- Migrate UI
+
+{% /callout %}
+
+Welcome to Nx 21 Launch Week! Nx 21 is here with some of the highest-impact changes we've ever built. We're really excited to share what we've been cooking up, so we'll be following up with more details throughout the week.
+
+{% toc /%}
+
+## ICYMI: Updates from Nx 20
+
+Before we dive into the newest features of Nx 21, make sure you catch these updates that happened during Nx 20:
+
+- [TypeScript Project References](/blog/new-nx-experience-for-typescript-monorepos) for Monorepos
+- [`@nx/rspack` graduates from labs](/blog/announcing-nx-20#nxrspack-graduates-from-labs)
+- [@nx/rspack-angular enables rspack builds for Angular](/blog/using-rspack-with-angular)
+- [nx import](/blog/announcing-nx-20#importing-existing-projects) allows you to import existing projects into an Nx workspace
+- [Local caching is now database-driven](/blog/announcing-nx-20#caching-now-with-databases)
+- [MCP server released](/blog/nx-mcp-vscode-copilot)
+
+## Continuous tasks
+
+During the development of Nx 20, we started an [RFC for continuous tasks](https://github.com/nrwl/nx/discussions/29025). After gathering feedback from the community, continuous tasks are here to make many workflows in your workspace much easier to implement.
+
+Many of the tasks in your workspace are finite: they run, produce an output, and shut down on their own. **Continuous tasks** are long-lived tasks: they run until interrupted by an outside input. These are tasks like serving your application or running tests in watch mode. While Nx has always supported running these tasks, you couldn't [configure a task pipeline](/concepts/task-pipeline-configuration). For example, you could serve your backend and frontend separately, but you couldn't easily configure your backend to be served whenever your frontend is served. There are always options like opening two separate terminals to run the tasks or setting up a specific script or task for running these in parallel. But the DX has always been lacking.
+
+Now, tasks can be marked as continuous, and other tasks can depend on them. Nx will no longer wait for these tasks to shut down before invoking the tasks that depend on them. Let's wire one up as an example: a frontend that depends on a backend running to function. Assuming we run a `dev` target from our `frontend` project, and a `serve` target from our `api` project, we configure this on the frontend project like this:
+
+```json {% fileName="apps/myfrontendapp/package.json" %}
+{
+  "nx": {
+    "name": "frontend",
+    "targets": {
+      "dev": {
+        "dependsOn": [{ "projects": ["api"], "target": "serve" }]
```

Comment:
I believe you also need to have some sort of dependency between `frontend -> api` in this case. Like an implicitDependency 🤔 . Need to check to be 100% sure, but afaik that is required

Reply:
I haven't needed one in my testing, but I'll double-check that it's not being added somehow

Reply:
In my test repo, I don't have any project dependency between frontend and api. Just the task dependency. confirmed this using the project graph and task graphs.

Is this expected? I don't see it mentioned in any docs of continuous tasks: https://canary.nx.dev/reference/project-configuration#continuous